drivers/ruby/lib/net.rb

```diff
@@ -677,20 +677,40 @@ def init_socket
         context = create_context(@ssl_opts)
         @socket = OpenSSL::SSL::SSLSocket.new(@tcp_socket, context)
         @socket.sync_close = true
-        @socket.connect
+        socket_connect
```

Comment:
I'm a little confused by this logic.  Below we call `socket_connect` after `base_socket`, but here we call `base_socket`, then put that into an `SSLSocket`, then call `socket_connect` on that.  Before we followed the same code path here for `@tcp_socket` as we did for `@socket` below, why did that change?

(Also, just to double-check, did you test these changes with both SSL and non-SSL connections?)